tests/geocoder/test_geocoder.py

```diff
@@ -0,0 +1,403 @@
+			"101 test ave apartment 2",
+			"101 test ave Unit 3",
+			"101 test ave unit 4",
+			"101 test ave #5",
+			"101 test ave # 6",
+			"101 test ave Lot 7",
+			"101 test ave lot 8",
+			"101 test ave Room 9",
+			"101 test ave rm 10",
+			"101 test ave bldg 11",
+			"101 test ave Building 12",
+			"101 test ave Upper",
+			"101 test ave uppr",
+			"101 test ave upr",
+			"101 test ave Lower",
+			"101 test ave lwer",
+			"101 test ave lwr",
+			"101 test ave Suite 13",
+			"101 test ave ste 14",
+			"101 test ave Fl 15",
+			"101 test ave frnt",
+			"101 test ave Trailer 16",
+			"101 test ave trlr 17",
+			"101 test ave Dept 18",
+			"101 test ave dept 19",
+			"101 test ave     rm 10",
+			"101 test ave"
+		]
+		for test in tests:
+			if not gc.street_encode(test) == "101 test ave":
+				errors.append(test)
+
+		error_msg = f"""{len(errors)} failed... 
+		street_encode failed to trim secondary descriptors from:"""
+		assert not errors, error_msg
+
+
+# ###### Test GeoLocations Methods ######
+class TestGeocodeLocations:
+	"""Tests for methods in bbd.geocoder.GeocodeLocations """
+
+	#Setting up cross-test enviornment
+	#data_df_path = Path("test_data/data.json").resolve()
+	data_df_path = Path(__file__).parent / "test_data/data.json"
+	data_df_path = data_df_path.resolve()
+	assert data_df_path.exists()
+	#data_list_path = Path("test_data/data.txt").resolve()
+	data_list_path = Path(__file__).parent / "test_data/data.txt"
+	data_list_path = data_list_path.resolve()
+	assert data_list_path.exists()
+
+	#Making address_df for testing pd.DataFrame implementation
+	#address_df has Address, City, State, Zip5 columns
+	address_df = pd.read_json(data_df_path)
+
+	#Making address_list for testing list of str implementation 
+	with open(data_list_path,"r") as f:
+		address_list = f.read().splitlines()
+
+	#Making address_dict for testing list of dict implementation
+	with open(data_df_path, "r") as f:
+		address_dict = json.load(f)
+
+
+	def test_GeocodeLocatios_make_file_header(self, tmp_path):
+		"""Tests the ._make_file() and ._make_header() methods
+		for GeocodeLocations
+		"""
+		p = tmp_path/"test.csv"
+
+		gl = gc.GeocodeLocations(self.address_df, valid_email, p)
+
+		test = pd.read_csv(p, sep = "\t")
+
+		error_msg = "Saved File has improper header."
+		assert "latitude" in test.columns, error_msg
+		assert "longitude" in test.columns, error_msg
+		assert "address" in test.columns, error_msg
+		assert len(test.columns) <= 4, error_msg
+
+
+	def test_GeocodeLocations_run_one_batch(self, tmp_path):
+		"""Tests the .run() and associated methods for
+		GeocodeLocations.
+		"""
+		p = tmp_path/"test.csv"
+
+		gl = gc.GeocodeLocations(self.address_df, valid_email, p)
+		gl._set_test_geocoder()
+
+		gl.run()
+
+		test = pd.read_csv(p, sep = "\t")
+
+		assert not test.empty, "No results saved to disk."
+		assert not gl.locations.empty, "No results stored in object."
+
+		assert len(test) == 234, "Not all lines were saved to disk"
```

Comment:
It is not b/c it is bypassing the actual _geocoder with a generator that gives either `geopy.Point` for Evan's Hall, Berkeley, CA's or None once in awhile. That means its avoiding the 1sec delay between calls.

Reply:
Weird. So I checked the automated tests where it says the runtime was more than ~1min15sec (b/c looking back at pre- #10 the runtime would go up to about there anyway) and if you click through the actual tests each one has a runtime <1min and since they're run in parallel it should only a few seconds longer than the longest test but the total runtimes say 4-7+min.